src/backend/cpu/CMakeLists.txt

```diff
@@ -1,247 +1,273 @@
-
-    IF("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.8.4")
-        MESSAGE("Disabling CPU Async as GCC Version ${COMPILER_VERSION} has known issues.")
-        MESSAGE("CPU Backend will use Synchronous Calls")
-        OPTION(BUILD_CPU_ASYNC "Build CPU backend with ASYNC support" OFF)
-    ELSE()
-        OPTION(BUILD_CPU_ASYNC "Build CPU backend with ASYNC support" ON)
-    ENDIF()
-
-    CMAKE_POLICY(POP)
-ENDIF(NOT DEFINED BUILD_CPU_ASYNC)
-SET(USE_CPUID ON CACHE BOOL "Build with CPUID integration")
-MARK_AS_ADVANCED(USE_CPUID)
-
-if (USE_CPUID)
-  ADD_DEFINITIONS(-DUSE_CPUID=1)
-ELSE(USE_CPUID)
-  ADD_DEFINITIONS(-DUSE_CPUID=0)
-ENDIF(USE_CPUID)
-
-IF (NOT ${BUILD_CPU_ASYNC})
-    ADD_DEFINITIONS(-DAF_DISABLE_CPU_ASYNC)
-ENDIF()
-
-IF(USE_CPU_F77_BLAS)
-    MESSAGE("Using F77 BLAS")
-    ADD_DEFINITIONS(-DUSE_F77_BLAS)
-ENDIF()
-
-IF(USE_CPU_MKL) # Manual MKL Setup
-    MESSAGE("CPU Backend Using MKL")
-    ADD_DEFINITIONS(-DUSE_MKL)
-ELSE(USE_CPU_MKL)
-    IF(${MKL_FOUND}) # Automatic MKL Setup from BLAS
-        MESSAGE("CPU Backend Using MKL RT")
-        ADD_DEFINITIONS(-DUSE_MKL)
-    ENDIF()
-ENDIF()
-
-IF (NOT CBLAS_LIBRARIES)
-    MESSAGE(SEND_ERROR "CBLAS Library not set")
-ENDIF()
-
-IF(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU" AND "${APPLE}")
-    ADD_DEFINITIONS(-flax-vector-conversions)
-ENDIF()
-
-FIND_PACKAGE(FFTW REQUIRED)
-
-IF(APPLE)
-    FIND_PACKAGE(LAPACKE QUIET) # For finding MKL
-    IF(NOT LAPACK_FOUND)
-        # UNSET THE VARIABLES FROM LAPACKE
-        UNSET(LAPACKE_LIB CACHE)
-        UNSET(LAPACK_LIB CACHE)
-        UNSET(LAPACKE_INCLUDES CACHE)
-        UNSET(LAPACKE_ROOT_DIR CACHE)
-        FIND_PACKAGE(LAPACK)
-    ENDIF()
-ELSE(APPLE) # Linux and Windows
-    FIND_PACKAGE(LAPACKE)
-ENDIF(APPLE)
-
-IF(NOT LAPACK_FOUND)
-    MESSAGE(WARNING "LAPACK not found. Functionality will be disabled")
-ELSE(NOT LAPACK_FOUND)
-    ADD_DEFINITIONS(-DWITH_CPU_LINEAR_ALGEBRA)
-ENDIF()
-
-IF(NOT UNIX)
-    ADD_DEFINITIONS(-DAFDLL)
-ENDIF()
-
-SET(THREADS_SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/threads")
-IF(EXISTS "${THREADS_SRC_DIR}" AND IS_DIRECTORY "${THREADS_SRC_DIR}"
-        AND EXISTS "${THREADS_SRC_DIR}/LICENSE")
-    # threads submodule has been initialized
-    # Nothing to do
-ELSE()
-    MESSAGE(STATUS "threads submodule unavailable. Updating submodules.")
-    EXECUTE_PROCESS(
-        COMMAND git submodule update --init --recursive
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-        )
-ENDIF()
-
-INCLUDE_DIRECTORIES(
-    ${CMAKE_INCLUDE_PATH}
-    "${PROJECT_SOURCE_DIR}/src/backend/cpu"
-    "${PROJECT_SOURCE_DIR}/src/backend/cpu/threads"
-    ${FFTW_INCLUDES}
-    ${CBLAS_INCLUDE_DIR}
-    )
-
-IF(LAPACK_FOUND)
-    INCLUDE_DIRECTORIES(${LAPACK_INCLUDE_DIR})
-ENDIF()
-
-FILE(GLOB cpu_headers
-    "*.hpp"
-    "*.h")
-
-FILE(GLOB cpu_sources
-    "*.cpp")
-
-LIST(SORT cpu_headers)
-LIST(SORT cpu_sources)
-
-source_group(backend\\cpu\\Headers FILES ${cpu_headers})
```

Comment:
`source_group` commands arranges the files in visually understandable hierarchy in Visual Studio projects. We have to bring this command back. These commands should be used in all places files are used for the above mentioned purpose.